### PR TITLE
Guard against clock manipulation by fake timer libraries. Fixes #128.

### LINKED ISSUE #128: Time is not calculated correctly when using a fake clock
I use [sinonjs](http://sinonjs.org/) for stubbing and stuff. But also to use a [fake timer](http://sinonjs.org/docs/#clock) that I can manipulate.

When doing this, times are not calculated correctly.

screen shot 2015-11-04 at 14 51 42

Will have a look at it myself, not sure when though.

### PR DESCRIPTION
**Pull request by jamestalmage**
Fixes #128

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 'use strict';
-var setImmediate = require('set-immediate-shim');
+var globals = require('./lib/globals');
```
Comment (sindresorhus) [inline index.js]: Can you place it with the other relative requires?

File: test/fork.js [hunk 2]
```diff
@@ -59,3 +59,12 @@ test('exit after tests are finished', function (t) {
 			cleanupCompleted = event.completed;
 		});
 });
+
+test('fake timers do not break duration', function (t) {
+	fork(fixture('fake-timers.js'))
+		.then(function (info) {
+			var duration = info.tests[0].duration;
+			t.true(duration < 1000, duration + ' < 1000');
```
Comment (sindresorhus) [inline test/fork.js]: Wish we had `power-assert` here so we didn't need the `duration + ' < 1000'`.

Comment (jamestalmage) [inline test/fork.js]: Yep. I'm so tempted to go create tap support. We will self test someday, and then all will be right with the world.

**Comment by jamestalmage** [comment]
@sindresorhus - made the change.

File: test/fork.js (hunk 2 above)
Comment (jprichardson) [inline test/fork.js]: > Yep. I'm so tempted to go create tap support. 

Pretty please, with sugar on top :blush:

Comment (jamestalmage) [inline test/fork.js]: Only so much time in the day. Don't hold your breath. :relaxed:

**Comment by sindresorhus** [comment]
:+1:

**Comment by jamestalmage** [comment]
The only hole I still see in this is someone using bluebird [`setScheduler`](http://bluebirdjs.com/docs/api/promise.setscheduler.html) and mucking things up.

**Comment by sindresorhus** [comment]
@jamestalmage Too much of an edge-case to care about IMHO.

**Comment by jamestalmage** [comment]
I don't think it's likely to be an issue until we actually start supporting the Browser. That angular example is literally the only way I've ever seen `setScheduler` used.

**Pull request merged**